projects/union_path_integration/convergence_simulation.py

```diff
@@ -28,9 +28,19 @@
 import io
 import os
 import random
+from multiprocessing import cpu_count, Pool
+from copy import copy
+import time
+try:
+  import cPickle as pkl
```

Comment:
change to json. Pickle has a lot of problems and for small files like this it is better to have human readable formats